src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -1065,6 +1068,24 @@ public String getChannelName() {
     return this.channel.getFullyQualifiedName();
   }
 
+  // ------ SETTERS ------ //
+  private void setOffsetPersistedInSnowflake(long offsetPersistedInSnowflake) {
+    this.offsetPersistedInSnowflake.set(offsetPersistedInSnowflake);
+    this.snowflakeTelemetryChannelStatus.setOffsetPersistedInSnowflake(offsetPersistedInSnowflake);
```

Comment:
Instead of getting the offset from here, why we decide to let the telemetry service maintain a copy of all three offsets?

Reply:
you're right, the SnowflakeTelemetryPipeStatus creates a copy, however that is not strictly necessary, so i've passed the offsets in TopicPartitionChannel creation